trunner/target/sparcv8.py

```diff
@@ -0,0 +1,133 @@
+import subprocess
+import time
+from typing import Callable, Optional
+
+import pexpect
+import pexpect.fdpexpect
+import usb.core
+
+from trunner.dut import SerialDut
+from trunner.harness import (
+    HarnessBuilder,
+    PloPhoenixdAppLoader,
+    PloHarness,
+    Rebooter,
+    ShellHarness,
+    FlashError,
+    TestStartRunningHarness,
+)
+from trunner.host import Host
+from trunner.tools import Phoenixd
+from trunner.types import TestResult, TestOptions
+from .base import TargetBase, find_port
+
+
+class GR716Rebooter(Rebooter):
+    def _reboot_soft(self):
+        self._reboot_hard()
+        # TODO implement soft restarts after implementing flash memory
+        # self.host.set_reset(1)
+        # time.sleep(0.5)
+        # self.dut.clear_buffer()
+        # self.host.set_reset(0)
+        # time.sleep(0.25)
+
+    def _reboot_hard(self):
+        # Make sure that that reset pin is in low state
+        self.host.set_reset(0)
+        self.host.set_power(False)
+        time.sleep(0.75)
+        self.host.set_power(True)
+        time.sleep(0.5)
+
+
+class GR716Target(TargetBase):
+    def __init__(self, host: Host, port: str, baudrate: int = 115200):
+        self.dut = SerialDut(port, baudrate, encoding="utf-8", codec_errors="ignore")
+        self.rebooter = GR716Rebooter(host, self.dut)
+        super().__init__()
+
+    @classmethod
+    def from_context(cls, ctx):
+        return cls(ctx.host, ctx.port, ctx.baudrate)
+
+    def MimasCustomUpload(self, boot_dir: str, image: str):
+        # using usb.core to fully reset device responsible for phoenixd
+        usb.core.find(idVendor=0x067B, idProduct=0x2303).reset()
+
+        fd = open(f"{boot_dir}/{image}", "rb")
+        self.rebooter()
+        filebuff = fd.read()
+
+        phoenix_process = subprocess.Popen(
+            f"phoenixd -p {find_port('067b:2303')} -b 115200 -s {str(boot_dir)}/.",
```

Comment:
If 067b and 2303 are repeating, it might be better to pass them directly to the function and use them